### ISSUE TITLE
UnhandledException displays when the download speed is 0b/s for a certain period of time

### ISSUE THREAD

**Issue by wutipong** [IllusionMods/KKManager#39]
Sometime, an `UnhandledException` dialog displays when the download starts, or when downloads have started and the network disconnected or briefly suspended, etc.

![image](https://user-images.githubusercontent.com/3213191/101584079-65b1be80-3a0f-11eb-8a62-4adfdf1cfe64.png)

This can be reproduced by starting the download, and then disconnect the network connection. After a while the dialog will display.

The problem is in the file `KKManager.Updater/Windows/ModUpdateProgressDialog.cs`, line 202-207, as of the time this issue report is written.  This claus `(double)averageDownloadSpeed.GetAverage()` becomes 0 and thus the `eta` is now positive infinity. `TimeSpan` is not smart enough to knows that there is a timespan call infinity, so it instead throws an `OverflowException`.

````C#
averageDownloadSpeed.Sample(downloadedSinceLast.GetKbSize());
var etaSeconds = (_overallSize - _completedSize).GetKbSize() / (double)averageDownloadSpeed.GetAverage();
var eta = TimeSpan.FromSeconds(etaSeconds);

labelPercent.Text =
$"Overall: {totalPercent:F1}% done  ({_completedSize} out of {_overallSize})\r\n" +
$"Speed: {downloadedSinceLast}/s  (ETA: {eta.GetReadableTimespan()})";
````

The solution would be not calculating ETA at all when the `averageDownloadSpeed.GetAverage()` is 0. For example :-

````C#
var averageDownloadSpeedValue = (double)averageDownloadSpeed.GetAverage();
if (averageDownloadSpeedValue == 0)
{
    labelPercent.Text =
        $"Overall: {totalPercent:F1}% done  ({_completedSize} out of {_overallSize})\r\n" +
        $"Speed: {downloadedSinceLast}/s";
}
else
{
    var etaSeconds = (_overallSize - _completedSize).GetKbSize() / (double)averageDownloadSpeed.GetAverage();
    var eta = TimeSpan.FromSeconds(etaSeconds);

    labelPercent.Text =
        $"Overall: {totalPercent:F1}% done  ({_completedSize} out of {_overallSize})\r\n" +
        $"Speed: {downloadedSinceLast}/s  (ETA: {eta.GetReadableTimespan()})";
        //$"Speed: {downloadedSinceLast:F1}KB/s";
}
````
and it would look like this when the speed is zero. The ETA will be back when the download comes back.

![image](https://user-images.githubusercontent.com/3213191/101584723-e7561c00-3a10-11eb-8a21-5fb9ef7bcfca.png)

PS. I decided not to send in a patch as I can't think of a good name for the variable .... `averageDownloadSpeedValue `. The name I used seems silly. May be you guys can figure out a better name or even better codes than what I can do.

**Comment by wutipong** [comment]
Just in case you might wonder why this dialog displays when this code block is in a try-catch block. The whole block is in a lambda so the outside try-catch block has no effects on it.

**Comment by ManlyMarco** [comment]
Fixed in 19f7710ae80027b5482f709ad89d4384e93074b5
Thanks for the report.
